src/vsphere_cpi/lib/cloud/vsphere/network.rb

```diff
@@ -58,7 +58,10 @@ def destroy(switch_id)
       switch = @switch_provider.get_switch_by_id(switch_id)
       t1_router_ids = @router_provider.get_attached_router_ids(switch_id)
       raise "Expected switch #{switch_id} to have one router attached. Found #{t1_router_ids.length}" if t1_router_ids.length != 1
-      switch_ports = @switch_provider.get_attached_switch_ports(switch_id)
+      switch_ports = @switch_provider.get_attached_switch_ports(switch_id).select do |logical_port|
+        @switch_provider.get_logical_port_status(logical_port.id) == "UP"
+      end
+
       raise "Expected switch #{switch_id} to have only one port. Got #{switch_ports.length}" if switch_ports.length != 1
```

Comment:
can you change the raised error message to say *found more than 1 operational switch ports or something like that*?